File: frontend/types/analysis.ts
export type AnalysisSeverity = "high" | "medium" | "low";

export interface DimensionScore {
  key: string;
  score: number;
  explanation: string;
  disclaimer?: string | null;
}

export interface AnalysisIssue {
  severity: string;
  category: string;
  title: string;
  description: string;
  suggested_fix?: string | null;
  grounded_in_resume?: boolean;
}

export interface ResumeAnalysis {
  analysis_id: string;
  resume_id: string;
  resume_version_id?: string | null;
  cached: boolean;
  stale?: boolean;
  overall_score: number | null;
  category_scores: Record<string, number>;
  summary: string;
  dimensions: DimensionScore[];
  issues: AnalysisIssue[];
  status: string;
  analyzed_at: string | null;
  prompt_version: string;
}

export interface LatestJobMatchSummary {
  match_id: string;
  job_description_id: string;
  job_title: string | null;
  company: string | null;
  match_score: number;
  semantic_score: number | null;
  breakdown: {
    skills_match?: number;
    experience_match?: number;
    keyword_match?: number;
    project_relevance?: number;
    education_match?: number;
  } | null;
  summary?: string;
  matched_at: string | null;
  stale?: boolean;
}

export interface ResumeDetail {
  id: string;
  user_id: string | null;
  title: string;
  original_filename: string | null;
  mime_type: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  sections_found: string[];
  sections_missing: string[];
  active_version_id?: string | null;
  active_version_label?: string | null;
  analysis_stale?: boolean;
  match_stale?: boolean;
  reanalyze_recommended?: boolean;
  latest_analysis: ResumeAnalysis | null;
  latest_job_match: LatestJobMatchSummary | null;
}

export interface ResumeListItem {
  id: string;
  title: string;
  original_filename: string | null;
  created_at: string;
  updated_at: string;
  overall_score: number | null;
  analyzed_at: string | null;
  has_analysis: boolean;
}

export const CATEGORY_CONFIG = [
  {
    key: "ats",
    label: "ATS Compatibility",
    dimensionKey: "ats_compatibility",
  },
  {
    key: "job_match",
    label: "Job Match",
    dimensionKey: null,
  },
  {
    key: "skills",
    label: "Skills",
    dimensionKey: "skills",
  },
  {
    key: "content_quality",
    label: "Content Quality",
    dimensionKey: "content_quality",
  },
  {
    key: "projects",
    label: "Projects",
    dimensionKey: "projects",
  },
  {
    key: "experience",
    label: "Experience",
    dimensionKey: "experience",
  },
] as const;

export const DIMENSION_LABELS: Record<string, string> = {
  ats_compatibility: "ATS Compatibility",
  skills: "Skills",
  experience: "Experience",
  projects: "Projects",
  education: "Education",
  certifications: "Certifications",
  achievements: "Achievements",
  professional_summary: "Professional Summary",
  keywords: "Keywords",
  content_quality: "Content Quality",
  readability: "Readability",
  relevance: "Relevance",
  quantifiable_achievements: "Quantifiable Achievements",
  action_verbs: "Action Verbs",
  formatting_issues: "Formatting",
};

export function normalizeSeverity(severity: string): AnalysisSeverity {
  const value = severity.toLowerCase();
  if (value === "high") return "high";
  if (value === "low") return "low";
  return "medium";
}

File: backend/app/schemas/analysis.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import Field

from app.models.enums import AnalysisStatus
from app.schemas.common import BaseSchema, TimestampSchema

# Forward reference avoided — job match summary on resume detail
class LatestJobMatchSummary(BaseSchema):
    match_id: uuid.UUID
    job_description_id: uuid.UUID
    job_title: str | None = None
    company: str | None = None
    match_score: int
    semantic_score: float | None = None
    breakdown: dict[str, Any] | None = None
    summary: str = ""
    matched_at: datetime | None = None
    stale: bool = False


class DimensionScoreRead(BaseSchema):
    key: str
    score: int
    explanation: str
    disclaimer: str | None = None


class AnalysisIssueRead(BaseSchema):
    severity: str
    category: str
    title: str
    description: str
    suggested_fix: str | None = None
    grounded_in_resume: bool = True


class AnalyzeResumeResponse(BaseSchema):
    analysis_id: uuid.UUID
    resume_id: uuid.UUID
    resume_version_id: uuid.UUID | None = None
    cached: bool
    stale: bool = False
    overall_score: int | None
    category_scores: dict[str, int]
    summary: str
    dimensions: list[DimensionScoreRead]
    issues: list[AnalysisIssueRead]
    status: str
    analyzed_at: datetime | None
    prompt_version: str


class ResumeAnalysisCreate(BaseSchema):
    resume_id: uuid.UUID
    resume_version_id: uuid.UUID | None = None
    overall_score: int | None = Field(default=None, ge=0, le=100)
    category_scores: dict[str, Any] | None = None
    issues: list[Any] | None = None
    status: AnalysisStatus = AnalysisStatus.PENDING
    analyzed_at: datetime | None = None


class ResumeAnalysisUpdate(BaseSchema):
    overall_score: int | None = Field(default=None, ge=0, le=100)
    category_scores: dict[str, Any] | None = None
    issues: list[Any] | None = None
    status: AnalysisStatus | None = None
    analyzed_at: datetime | None = None


class ResumeAnalysisRead(TimestampSchema):
    id: uuid.UUID
    resume_id: uuid.UUID
    resume_version_id: uuid.UUID | None
    overall_score: int | None
    category_scores: dict[str, Any] | None
    issues: list[Any] | None
    status: AnalysisStatus
    analyzed_at: datetime | None


class ResumeDetailResponse(BaseSchema):
    id: uuid.UUID
    user_id: uuid.UUID | None
    title: str
    original_filename: str | None
    mime_type: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime
    sections_found: list[str]
    sections_missing: list[str]
    active_version_id: uuid.UUID | None = None
    active_version_label: str | None = None
    analysis_stale: bool = False
    match_stale: bool = False
    reanalyze_recommended: bool = False
    latest_analysis: AnalyzeResumeResponse | None = None
    latest_job_match: LatestJobMatchSummary | None = None
